### PR TITLE
doc: update plugin-extensibility.md spec

### LINKED ISSUE #190: Update directory structure related specs
According to [directory structure spec](https://github.com/notaryproject/notation/blob/main/specs/directory.md), documents related to directory structure should be updated.
1. [plugin-extensibility](https://github.com/notaryproject/notaryproject/blob/cose-envelope/specs/plugin-extensibility.md) spec
2. [hello-signing](https://github.com/notaryproject/notation/blob/main/docs/hello-signing.md)

### PR DESCRIPTION
**Pull request by JeyJeyGao**
According to directory structure spec in notation repo, the plugin
directory should be updated.

Resolves #190 
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Comment by priteshbandi** [comment]
@JeyJeyGao Can you please update the PR to only use user directory for plugin?

cc: @shizhMSFT

**Comment by JeyJeyGao** [comment]
> @JeyJeyGao Can you please update the PR to only use user directory for plugin?
> 
> cc: @shizhMSFT

Hi @priteshbandi, thanks for reminding me. Updated.

File: specs/plugin-extensibility.md [hunk 1]
```diff
@@ -30,16 +30,16 @@ Plugin publisher will provide instructions to download and install the plugin. P
...
+| ------- | ---------------------------------------------------- |
+| Unix    | `$XDG_CONFIG_HOME/notation/plugins`                  |
+| Windows | `%AppData%/notation/plugins`                         |
+| Darwin  | `$HOME/Library/Application Support/notation/plugins` |
 
-* Unix-like OSes:
-  * `$HOME/notation/plugins`
-* On Windows:
-  * `%USERPROFILE%\notation\plugins`
+Each plugin executable and dependencies are installed under directory `$PLUGIN_DIRECTORY/{plugin-name}` with an executable under that directory `$PLUGIN_DIRECTORY/{plugin-name}/notation-{plugin-name}`.
```
Comment (priteshbandi) [inline specs/plugin-extensibility.md]: we dont care where dependencies are installed.
```suggestion
Each plugin executable must be located under `$PLUGIN_DIRECTORY/{plugin-name}` directory, with executable named as  `notation-{plugin-name}`.
```

**Review by priteshbandi — COMMENTED** [review]
One comment else LGTM

File: specs/plugin-extensibility.md (hunk 1 above)
Comment (JeyJeyGao) [reply specs/plugin-extensibility.md]: Updated.

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**